### PR TITLE
Add Legacy CD Keys Dumper Example

### PR DESCRIPTION
**Pull request by Revadike**
Original: #390

### CODE REVIEW

File: examples/legacycdkeysdumper.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            Object.keys(client.picsCache?.apps[appid]?.appinfo?.extended || {}).map(k => k.toLowerCase()).includes(key)
+        );
+    });
+
+
+    console.log("Requesting legacy cd keys for " + apps.length + " owned apps...");
+    let keys = {};
+    for (let appid of apps) {
+        let { key } = await client.getLegacyGameKey(appid).catch(err => {
+            if (err.eresult !== 2) { // usually just means no cd key present
```
Comment (DoctorMcKay) [inline examples/legacycdkeysdumper.js:41]: Should explicitly compare against the `EResult` enum. An example should follow best practices and teach people how to use enums.

File: examples/legacycdkeysdumper.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        ].some(key =>
+            Object.keys(client.picsCache?.apps[appid]?.appinfo?.extended || {}).map(k => k.toLowerCase()).includes(key)
+        );
+    });
+
+
+    console.log("Requesting legacy cd keys for " + apps.length + " owned apps...");
+    let keys = {};
+    for (let appid of apps) {
+        let { key } = await client.getLegacyGameKey(appid).catch(err => {
```
Comment (DoctorMcKay) [inline examples/legacycdkeysdumper.js:40]: Please use try/catch if you're awaiting the promise.

File: examples/legacycdkeysdumper.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            "legacykeylinkedexternally",
+            "legacykeyproofofpurchaseticket",
+            "legacykeyregistrationmethod",
+            "legacykeyregistrylocation",
+            "showcdkeyinmenu",
+            "showcdkeyonlaunch",
+            "supportscdkeycopytoclipboard",
+            "thirdpartycdkey",
+        ].some(key =>
+            Object.keys(client.picsCache?.apps[appid]?.appinfo?.extended || {}).map(k => k.toLowerCase()).includes(key)
```
Comment (DoctorMcKay) [inline examples/legacycdkeysdumper.js:32]: Optional chaining is not available prior to node 13.2.0, and the minimum node version in package.json is 8.0.0. We shouldn't use features not available in the minimum supported node version.

File: examples/legacycdkeysdumper.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        );
+    });
+
+
+    console.log("Requesting legacy cd keys for " + apps.length + " owned apps...");
```
Comment (DoctorMcKay) [inline examples/legacycdkeysdumper.js:37]: I think I'd like to see a template literal used here, but not a big deal.

File: examples/legacycdkeysdumper.js (hunk 2 above)
Comment (Revadike) [reply examples/legacycdkeysdumper.js:40]: Wait, why.

Comment (DoctorMcKay) [reply examples/legacycdkeysdumper.js:40]: The entire point of async/await; using the .catch method seems to defeat the purpose to me.

**Comment by Revadike** [comment]
Requested changes implemented.
Requesting a new code review from @DoctorMcKay.

**Pull request merged**